src/format/parse.rs

```diff
@@ -224,6 +224,103 @@ pub(crate) fn parse_rfc3339<'a>(parsed: &mut Parsed, mut s: &'a str) -> ParseRes
     Ok((s, ()))
 }
 
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub(crate) enum Iso8601Format {
+    Basic,
+    Extended,
+    #[allow(unused)]
+    Unknown,
+}
+
+/// The ISO 8601 date format is a combination of 12 different date formats:
+///
+/// |                                           | calendar date | ordinal date |  week date   |
+/// |-------------------------------------------|---------------|--------------|--------------|
+/// | basic format                              | YYYYMMDD      | YYYYDDD      | YYYYWwwD     |
+/// | extended format                           | YYYY-MM-DD    | YYYY-DDD     | YYYY-Www-D   |
+/// | basic format, expanded representations    | ±Y̲YYYYMMDD    | ±Y̲YYYYDDD    | ±Y̲YYYYWwwD   |
+/// | extended format, expanded representations | ±Y̲YYYY-MM-DD  | ±Y̲YYYY-DDD   | ±Y̲YYYY-Www-D |
+///
+/// Returns `(remainder, Iso8601Format)`.
+//// - The ISO 8601 format of the date is returned so the calling function can check it matches the
+///   format of a time component (basic or extended format).
+pub(crate) fn parse_iso8601_date<'a>(
+    parsed: &mut Parsed,
+    mut s: &'a str,
+) -> ParseResult<(&'a str, Iso8601Format)> {
```

Comment:
To reuse this method for parsing a duration it should take a flag to indicate it should not parse *week date* formats.